### PR TITLE
Major optimizations

### PR DESCRIPTION
**Pull request by seiflotfy**
* map/reduce all incoming add values of a request into a map
* implement BulkAdd for all sketches
* use new count-min-log
* additions to domains are done in go routines
* add ```make bench``` command

### CODE REVIEW

File: src/manager/manager_test.go [hunk 1]
```diff
@@ -381,5 +381,6 @@ func TestCreateDeleteDomain(t *testing.T) {
 		t.Error("Expected [[dc card]], got", sketches[0][0], sketches[0][1])
 	} else if sketches[1][0] != "dc" || sketches[1][1] != "freq" {
 		t.Error("Expected [[dc freq]], got", sketches[1][0], sketches[1][1])
+		fmt.Println(sketches)
```
Comment (martinpinto) [inline src/manager/manager_test.go]: we should probably use loggo here

Comment (seiflotfy) [inline src/manager/manager_test.go]: nope we need to get rid of this line. good call
no print outs in tests unless they are fails raised by t.Error

File: src/manager/domain.go [hunk 2]
```diff
@@ -90,13 +91,24 @@ func (m *domainManager) delete(id string) error {
...
+	var wg sync.WaitGroup
+	wg.Add(len(sketches))
+
 	for _, sketch := range sketches {
-		_ = m.sketches.add(sketch, values)
-		// fmt.Println(err)
+		go func(sk string) {
+			defer wg.Done()
+			if err := m.sketches.add(sk, values); err != nil {
+				fmt.Println(err)
```
Comment (njpatel) [inline src/manager/domain.go]: loggo

File: src/manager/domain.go [hunk 3]
```diff
@@ -90,13 +91,24 @@ func (m *domainManager) delete(id string) error {
...
 	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(sketches))
+
 	for _, sketch := range sketches {
-		_ = m.sketches.add(sketch, values)
-		// fmt.Println(err)
+		go func(sk string) {
+			defer wg.Done()
```
Comment (njpatel) [inline src/manager/domain.go]: don't use defer in tight loops, better to just call it at the bottom

Comment (seiflotfy) [inline src/manager/domain.go]: if something fails here. the wait in the bottom never terminates. right?

File: src/server/server.go [hunk 4]
```diff
@@ -28,6 +29,8 @@ var server *serverStruct
 
 // Run ...
 func Run(manager *manager.Manager, port uint) {
+	nCPU := runtime.NumCPU()
+	runtime.GOMAXPROCS(nCPU)
```
Comment (njpatel) [inline src/server/server.go]: is this still necessary with 1.5+ ?

File: src/manager/domain.go (hunk 3 above)
Comment (njpatel) [inline src/manager/domain.go]: if it segfaults or panics, it'll cause skizze to crash anyway right

**Comment by njpatel** [comment]
Couple notes, otherwise :+1:

File: src/server/server.go (hunk 4 above)
Comment (seiflotfy) [inline src/server/server.go]: Yes as far as I can tell.

**Pull request merged**